docs/livewire.md

```diff
@@ -0,0 +1,188 @@
+## A brief introduction to Livewire
+
+We first give a short intuition of how Livewire, the framework of the new front-end works.
+
+### Livewire and Controllers
+
+The Laravel framework on which Lychee is based provides to options how a request may be routed:
+
+- the request is directed to a controller below `app/Http/Controllers`
+- the request is directed to a Livewire component below `app/Http/Livewire`
+
+In the first case, a controller returns a JSON object or a HTML page rendered from a blade template.
+In the second case (Livewire), only a block of HTML built from a blade template will be returned.
+
+From a very high level Livewire makes use of two folders:
+
+- `app/Http/Livewire` which hosts the components and their inner logic.
+- `resources/views/components` which hosts the HTML layout of the components.
+
+So far one could say there are not much difference with a normal request life cycle. Where Livewire starts to shine is in the ability to call the methods of the component directly _from the HTML_. Instead of having to use a JavaScript hook, make an AJAX call, parse the result, and update the layout accordingly; the user/programmer can directly interact with the designated component and Livewire takes care of the request and rendering.
```

Comment:
```suggestion
So far one could say there is not much difference with a normal request life cycle. Where Livewire starts to shine is in the ability to call the methods of the component directly _from HTML_. Instead of having to use a JavaScript hook, make an AJAX call, parse the result, and update the layout accordingly, the user/programmer can directly interact with the designated component and Livewire takes care of the request and rendering.
```